server/game/core/card/CardTypes.ts

```diff
@@ -57,6 +57,10 @@ export type PlayableCard =
     UpgradeCard |
     NonLeaderUnitCard;
 
+/** Type union for any card type that can enter play (with deploy or played) */
+export type EnteredPlayCard =
```

Comment:
This concept already exists, it's `InPlayCard` which covers upgrades, units, and leaders.